nvflare/fuel/utils/buffer_list.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+class BufferList:
+    """A buffer list that can be treated as a single buffer"""
+
+    def __init__(self, buf_list: list):
+        self.buf_list = buf_list
+        if buf_list:
+            self.size = sum(len(buf) for buf in buf_list)
+        else:
+            self.size = 0
+
+    def get_size(self):
+        return self.size
+
+    def get_list(self):
+        return self.buf_list
+
+    def append(self, buf: bytes):
```

Comment:
Not sure if self.size needs to be updated when self.buf_list.append(buf).  From the init, self.size is the sum of the sizes of all buf in buf_list.  So once a new buf is added, self.size seems outdated.

Reply:
Good catch! The size calculation is changed to dynamic.